llms/mlx_lm/utils.py

```diff
@@ -498,7 +498,7 @@ def save_weights(
         shard_name = shard_file_format.format(i + 1, shards_count)
         shard_path = save_path / shard_name
 
-        mx.save_safetensors(str(shard_path), shard)
+        mx.save_safetensors(str(shard_path), shard, metadata={"format": "pt"})
```

Comment:
I'm wondering if this should be an option rather than the default. On the one hand MLX doesn't really need that field so it's kind of unused otherwise. On the other hand it seems a bit strange to default the format to PyTorch whenever saving MLX models. Also we may want to use that field in the future..

Do you all know where does the format gets read in Transformers? Maybe we could ask them to add a MLX field and default the format to MLX instead?

Reply:
Yeah, what you said actually makes a lot of sense. From the error I was getting, it looks like if the Transformers library added 'mlx' to the following line, and if MLX default saves safetensors files with format="mlx" this can be achieved.

https://github.com/huggingface/transformers/blob/63caa370e6c618dbe7d3fd4cbf545cc32eca1a15/src/transformers/modeling_utils.py#L507

Reply:
Would you be open to making an issue or PR for that and link it? I would definitely comment / +1 it.

Reply:
Yeah, sure thing.

Reply:
@awni I added an issue/PR on Transformers repo: 
- https://github.com/huggingface/transformers/issues/29333
- https://github.com/huggingface/transformers/pull/29335

I also made a commit to save the format in "mlx" instead per above discussion.

Reply:
Awesome, thanks!